pkg/api/v1beta3/dynakube/oneagent_props.go

```diff
@@ -77,30 +82,8 @@ func (dk *DynaKube) OneAgentConnectionInfoConfigMapName() string {
 	return dk.Name + OneAgentConnectionInfoConfigMapSuffix
 }
 
-func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
-	return (dk.HostMonitoringMode() || dk.CloudNativeFullstackMode()) &&
-		dk.FeatureReadOnlyOneAgent()
-}
-
-func (dk *DynaKube) IsAppMonitoringWithCSI() bool {
-	if !dk.ApplicationMonitoringMode() {
-		return false
-	}
-
-	defaultUseCSIDriver := false
-	if dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver == nil {
-		return defaultUseCSIDriver
-	}
-
-	return *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
-}
-
-func (dk *DynaKube) NeedsCSIDriver() bool {
-	isAppMonitoringWithCSI := dk.IsAppMonitoringWithCSI()
-
-	isHostMonitoringWithCSI := dk.HostMonitoringMode() && dk.FeatureReadOnlyOneAgent()
-
-	return dk.CloudNativeFullstackMode() || isAppMonitoringWithCSI || isHostMonitoringWithCSI
+func (dk *DynaKube) UseReadOnlyOneAgents() bool {
```

Comment:
it is used both in validation and daemonset generation, so yes
- this is mainly annoying due to having classic in the daemonset generation, otherwise it could be moved to the validation package

also `UseCSIVolume` is a bit too generic, as it does not consider the application monitoring (I used to have a check for that but you considered it too convoluted 😅)